docs/_Docs_Upgrades.md

```diff
@@ -1,27 +1,36 @@
+In other words, upgrades to any individual colony on the network are "opt-in", while the network as a whole remains eternally backwards-compatible.
 
-The contracts comprising the Colony Network are upgradeable using the design pattern called EtherRouter.
+## For whom the delegate calls
+Interacting with both the Colony Network and individual colonies on the network is somewhat different than many other smart contract interactions that a blockchain developer might not be accustomed to.
 
-## EtherRouter
-This pattern uses two contracts in addition to the contract(s) providing their intended functionality:
+Rather than calling functions directly from the contract in which they are deployed, all transactions are signed and sent to the `EtherRouter` contract.
 
-* The `EtherRouter` contract which passes transactions (via `delegatecall`) to the contract that implements the called function.
-* A `Resolver` contract where the addresses of the contracts that implement the desired function are defined.
+Whenever a transaction is received by the `EtherRouter` contract, it looks the function up in a `Resolver` contract, using the function signature.
 
-Whenever a transaction is received by the `EtherRouter` contract, it looks up the contract that implements that function (if any) in the `Resolver`, and then `delegatecall`s that contract.
+The `Resolver` contract contains a mapping of whitelisted function signatures to addresses (`mapping ( bytes4 => address )`).
+
+A function signature lookup will return the address of the contracts that implement the desired function. `EtherRouter` in turn calls the function via `delegatecall`, and passes any returns from the call back to `msg.sender`.
 
 ![EtherRouter](img/delegateProxyCallchain_1.png)
 
-In order to upgrade, new contracts are deployed with new functionality, and then contracts that the `Resolver` contract points to must be changed to point to these new contracts.
+This pattern enables both fine-grained control of permissions for individual functions (see below), well as eternal backwards-compatibility following network upgrades. To learn more about planned upgrades to the Colony Network, please see [Releases](/colonynetwork/docs-releases/).
+
+## It calls for `roleId`
+Most functions in colony are *authorized* by a separate contract which defines the rules for who can call which functions. Any function that is decorated with the `auth` modifier will perform an authorization check before granting access to the function.
+
+In the current Glider release, authority is based on roles, which are defined in the relevant "authority" contracts, e.g. `ColonyAuthority.sol`.
+
+Roles within Colony act as a white-list for functions, registering specific addresses to a `Founder` or `Admin` role, approved to call a certain set of functions within a colony.
```

Comment:
Would it be better to start describing the new domain-level roles? I.e. `Root`, `Architecture`, etc?